src/openforms/submissions/api/permissions.py

```diff
@@ -67,7 +68,14 @@ def has_object_permission(self, request: Request, view: APIView, obj) -> bool:
         submission_url_kwarg = getattr(view, "submission_url_kwarg", default_url_kwarg)
 
         submission_uuid = view.kwargs[submission_url_kwarg]
-        return owns_submission(request, submission_uuid)
+        submission = (
+            obj
+            if isinstance(obj, Submission)
+            else Submission.objects.get(uuid=submission_uuid)
```

Comment:
I think it's now technically possible for people to trigger `Submission.DoesNotExist` with carefully crafted URLs. I think you can guard against that with a guard clause:

```py
if not owns_submission(request, submission_uuid):
    return False
```

This gives stronger guarantees that the record actually exists in the DB and saves a (crashing) query when people are trying to poke holes.